src/audio/kpb.c

```diff
@@ -917,8 +970,10 @@ static int kpb_reset(struct comp_dev *dev)
 			kpb_reset_history_buffer(kpb->hd.c_hb);
 		}
 
+#ifndef CONFIG_AMS
 		/* Unregister KPB from notifications */
 		notifier_unregister(dev, NULL, NOTIFIER_ID_KPB_CLIENT_EVT);
+#endif
```

Comment:
no need to unregister AMS here?

Reply:
No, there is no need to unregister AMS here. `kpb_reset()` is called during `PIPELINE_RESET ` state. RESET on pipeline could be called in two cases:
1. Init -> Reset -> Pause -> Running
2. Pause -> Reset -> Delete

So calling `ams_helper_unregister_consumer()` in kpb_free() will be sufficient.

Reply:
then why is `notifier_unregister()` called both in reset and free? Is that superfluous?